### DISCUSSION TITLE
Service unresponsiveness when hosted on EC2

### DISCUSSION THREAD

**Discussion by badolamgk** [umputun/remark42#1851] [Q&A] [answered]
Hello, long time user of your project.

I have been facing a very peculiar issue which seems to keep coming back on a frequent basis. I run a custom fork of Remark42 on an AWS EC2 instance, which has `nginx` to handle requests from public internet to the Remark42 app. It chokes up somewhere, which leads to the EC2 instance not responding and all the request URLs throwing `Timeout (no headers received)` status. I have to then stop the instance and then start it, and then run `compose` to start the application again. Simply running the `Reboot Instance` option doesn't work.[1]

The fork doesn't have any custom processing logic, it is just some frontend changes and a new import command for adding MSN comments. There is no pattern found when the downtime have occurred and I have been scaling up instance type but it is still persisting.

To give context about traffic, the website has been having pageviews in the range of 500k-2M, and we use <iframe> tag to render the comment interface.

I need guidance as to how can I detect the issue or reduce the downtime. If I am using orchestration like Kubernetes, how can I ensure that the database statefulness is maintained across all nodes?

Thanks.

[1] Instance choking up at 17:04 PM
<img width="1035" alt="Instance choking up at 17:04 PM" src="https://github.com/user-attachments/assets/ae06ab6b-84fd-4051-a95a-3faf2aed58cf">

**Comment by umputun** [comment, marked as answer]
The troubleshooting should be done for the EC2 setup, not Remark42. You should probably reach out to AWS support and ask them to help with this diagnostic. From what I see in your monitoring graphs, it looks like you are using some T-family instances and are running out of credits. If this is the case, you can configure your instance to automatically increase the balance; it has an option for this.

**Reply by badolamgk** [reply]
oh okay, will raise a ticket there. Thanks.
